pkg/controllers/controllers.go

```diff
@@ -56,7 +56,7 @@ import (
 func NewControllers(ctx context.Context, mgr manager.Manager, sess *session.Session, clk clock.Clock, kubeClient client.Client, recorder events.Recorder,
 	unavailableOfferings *cache.UnavailableOfferings, cloudProvider cloudprovider.CloudProvider, subnetProvider subnet.Provider,
 	securityGroupProvider securitygroup.Provider, instanceProfileProvider instanceprofile.Provider, instanceProvider instance.Provider,
-	pricingProvider pricing.Provider, amiProvider amifamily.Provider, launchTemplateProvider launchtemplate.Provider, instanceTypeProvider instancetype.Provider) []controller.Controller {
+	pricingProvider pricing.Provider, amiProvider amifamily.Provider, launchTemplateProvider launchtemplate.Provider, instanceTypeProvider *instancetype.DefaultProvider) []controller.Controller {
```

Comment:
I see the technical reason further in the PR which the controller depends on, but should we reduce the scope?

Reply:
You're asking about the scope of the PR itself? This interface is technically non-sensical right now since, yes, receiving the `instancetype.Provider` into this controller will technically "work". This controller literally only works for the `DefaultProvider` because it access data members of that provider. If you were to pass another interface to it, it would fail.

You _could_ change the interface to be something like `SetInstanceTypeInfo` and `SetInstanceTypeOfferingInfo` but I'd rather just scope down the interface and say that these controllers specifically apply to this provider for now